src/components/EarnDashboard/components/ProviderDetails/OpportunityRow.tsx

```diff
@@ -0,0 +1,208 @@
+import { Button, Flex, List, useColorModeValue } from '@chakra-ui/react'
+import { DefiAction } from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import { useCallback, useMemo } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { Amount } from 'components/Amount/Amount'
+import { AssetCell } from 'components/StakingVaults/Cells'
+import { RawText } from 'components/Text'
+import { bn, bnOrZero } from 'lib/bignumber/bignumber'
+import type {
+  LpEarnOpportunityType,
+  StakingEarnOpportunityType,
+} from 'state/slices/opportunitiesSlice/types'
+import {
+  getRewardBalances,
+  getUnderlyingAssetIdsBalances,
+} from 'state/slices/opportunitiesSlice/utils'
+import {
+  selectAssetById,
+  selectAssets,
+  selectMarketDataSortedByMarketCap,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { NestedAsset } from './NestedAsset'
+import { OpportunityRowGrid } from './OpportunityTableHeader'
+
+type OpportunityRowProps<T extends StakingEarnOpportunityType | LpEarnOpportunityType> = {
+  onClick: (opportunity: T, action: DefiAction) => void
+  opportunity: T
+}
+
+export const OpportunityRow: React.FC<
+  OpportunityRowProps<StakingEarnOpportunityType | LpEarnOpportunityType>
+> = ({ onClick, opportunity }) => {
+  const {
+    assetId,
+    underlyingAssetId,
+    fiatAmount,
+    isClaimableRewards,
+    underlyingAssetRatiosBaseUnit,
+    cryptoAmountBaseUnit,
+    underlyingAssetIds,
+    type,
+    apy,
+    icons,
+  } = opportunity
+  const translate = useTranslate()
+  const borderColor = useColorModeValue('blackAlpha.50', 'whiteAlpha.50')
+  const asset = useAppSelector(state => selectAssetById(state, underlyingAssetId))
+  const assets = useAppSelector(selectAssets)
+  const marketData = useAppSelector(selectMarketDataSortedByMarketCap)
+
+  const underlyingAssetLabel = useMemo(() => {
+    if ((opportunity as StakingEarnOpportunityType)?.rewardsCryptoBaseUnit) {
+      return 'common.reward'
+    } else {
+      return 'defi.underlyingAsset'
+    }
+  }, [opportunity])
+
+  const underlyingAssetBalances = useMemo(() => {
+    if ((opportunity as StakingEarnOpportunityType)?.rewardsCryptoBaseUnit) {
+      const earnOpportunity = opportunity as StakingEarnOpportunityType
+      return getRewardBalances({
+        rewardAssetIds: earnOpportunity.rewardAssetIds,
+        rewardsCryptoBaseUnit: earnOpportunity.rewardsCryptoBaseUnit,
+        assets,
+        marketData,
+      })
+    } else {
```

Comment:
Can ditch the else and squirly braces if you feel like it
![squirlybraces](https://media.tenor.com/TwUsDToDjrIAAAAM/theprimeagen-surprised.gif)